requirements.txt

```diff
@@ -1,2 +1,3 @@
 beautifulsoup4
 behave
+django>=2.2
```

Comment:
This was intentionally left out in the past to avoid the known issues of a package that installs Django all by itself.

I leave it in for now, but we may discover that this was no good idea.

Reply:
I saw this constraint used in other high quality packages so I added also here.

I'm interested in the info about the known issues you mentioned before.

Reply:
I don't find the source now, but the gist of the problem is:

- When Django is a dependency of a package and Django is not installed yet (!) then installing the package will install the best-matching Django version (as in "just any Django version, which satisfies the dependency I specify as a package developer").
- This can lead to a conflict when that Django version is incompatible with the explicit requirements that you have specified in your project (your project's `requirements.txt` file or similar).

Hence, it's preferred to leave the responsibility of ensuring Django is installed to the (project) developer.

I'm interested in knowing which "high quality packages" (no pun intended) specify Django as an explicit dependency, and why or whether this is not a problem. Can you post a list here?

Here, in `behave-django` we have taken a similar decision related to the [dependency on `behave`](https://github.com/behave/behave-django/blob/master/requirements.txt#L2). The reason we did this, though, is twofold:

1. Running BDD tests is a development activity, and pulling in `behave` as a dependency is less critical (if it goes wrong) than pulling in a "wrong version" unnoticed on a customer project. And the reputation of the `behave-django` package is less at stake if it causes hick-ups due to a potential side-effect (compared to "feature-focused" packages).
2. On the convenience side, I feel I'd rather specify only `behave-django` as a development dependency than `behave-django` _and_ `behave`, which only clutters the dependency list. It's still possible to override this by installing `behave` first, explicitly.

Reply:
> ... it's preferred to leave the responsibility of ensuring Django is installed to the (project) developer.

I guess we want to prevent the project developer from using our package with a Django version that is no longer supported (e.g. Django 1.11 or 2.0), I would leave it to him to use compatible packages and versions.

> I'm interested in knowing which "high quality packages" (no pun intended) specify Django as an explicit dependency, and why or whether this is not a problem. Can you post a list here?

I'll list below only the first three examples I found:

- [django-rest-framework](https://github.com/encode/django-rest-framework/blob/master/setup.py#L85)
- [django-debug-toolbar](https://github.com/jazzband/django-debug-toolbar/blob/main/setup.cfg#L36)
- [django-configurations](https://github.com/jazzband/django-configurations/blob/master/setup.py#L30)